_opensearch/search-backpressure.md

```diff
@@ -0,0 +1,217 @@
+---
+layout: default
+title: Search backpressure
+nav_order: 63
+has_children: false
+---
+
+# Search backpressure
+
+Search backpressure is a mechanism to identify resource-intensive search requests and cancel them when the node is under duress. If a search request on a node or shard has breached the resource limits and does not recover within a certain threshold, it is rejected. These thresholds are dynamic and configurable through [cluster settings](#search-backpressure-settings). 
+
+## Measuring resource consumption
+
+To decide whether to apply search backpressure, OpenSearch periodically measures the following resource consumption statistics for each search request:
+
+- CPU usage
+- Heap usage
+- Elapsed time 
+
+An observer thread tracks the resource consumption of each task thread. It measures the resource consumption at several checkpoints during the query phase of a shard search request. If the node is determined to be under duress based on the JVM memory pressure and CPU utilization, the server examines the resource consumption for each search task. It determines if the CPU usage and elapsed time are within their fixed thresholds, and it compares the heap usage against the rolling average of the heap usage of the 100 most recent tasks. If the task is among the most resource-intensive based on these criteria, the task in canceled. 
+
+Every minute OpenSearch can cancel at most 1% of the number of currently running search shard tasks. Once a task is canceled, OpenSearch monitors the node for the next two seconds to determine if it is still under duress.
+
+## Canceled queries
+
+If a query is canceled, instead of receiving search results you receive an error from the server similar to the error below:
```

Comment:
Depending on what all shards failed, it may be possible that OpenSearch returns partial results. Can we re-word this slightly?